File: src/main/scala/us/zoom/data/dfence/providers/snowflake/models/opaque/SnowflakeGrantObjectName.scala
package us.zoom.data.dfence.providers.snowflake.models.opaque

opaque type SnowflakeGrantObjectName = String

object SnowflakeGrantObjectName:
  def apply(name: String): SnowflakeGrantObjectName                       = name.trim
  extension (grantObjectName: SnowflakeGrantObjectName) def toStr: String = grantObjectName.trim

File: src/main/java/us/zoom/data/dfence/providers/snowflake/revoke/models/wrappers/SnowflakeGrantObjectName.java
package us.zoom.data.dfence.providers.snowflake.revoke.models.wrappers;

public record SnowflakeGrantObjectName(String value) {
  public static SnowflakeGrantObjectName of(String value) {
    return new SnowflakeGrantObjectName(value.trim());
  }
}
